homeassistant/components/recorder/migration.py

```diff
@@ -530,10 +526,7 @@ def _modify_columns(
 
     if engine.dialect.name == SupportedDialect.POSTGRESQL:
         columns_def = [
-            "ALTER {column} TYPE {type}".format(
-                **dict(zip(["column", "type"], col_def.split(" ", 1), strict=False))
-            )
-            for col_def in columns_def
+            "ALTER " + " TYPE ".join(col_def.split(" ", 1)) for col_def in columns_def
```

Comment:
This significantly affects readability and I can't quite tell if it's correct or not. Maybe something like this:

```python
columns_def = [
    f"ALTER {column} TYPE {type}"
    for column, type in (col_def.split(" ", 1) for col_def in columns_def)
]
```